wordpress-mobile/WordPress-Android
# Implements: Viewmodel slice for ActivityLogCard

Closes #19022

## Description
Implements the viewmodel slice for Activity log card

## To test:
- Go to app 
- Login with a site having activity log 
- Go to me → Debug settings 
- Verify that `dashboard_activity_log` flag doesn’t exist 
- Click on an `ActivityCardItem`
- Verify that the event is triggered 
- ✅  🔵 Tracked: my_site_dashboard_card_item_tapped, Properties: {"type":"activity_log","subtype":"activity_log"}
- Navigate Back 
- Click on more menu 
- ✅  Verify that the event is triggered
-  🔵 Tracked: my_site_dashboard_contextual_menu_accessed, Properties: {"card":"activity_log"}
- Click on all activity
- ✅  Verify  🔵 Tracked: my_site_dashboard_card_menu_item_tapped, Properties: {"card":"activity_log","item":"all_activity"}
- Navigate back 
- Click on more menu 
- Click on Hide this 
- Verify event triggered 🔵 Tracked: my_site_dashboard_card_menu_item_tapped, Properties: {"card":"activity_log","item":"hide_this"}

## Regression Notes
1. Potential unintended areas of impact
- Activity log card does not work as expected
- Tracking on Activity log card is incorrect

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests and Unit tests

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:

UI Changes testing checklist:
